### PR TITLE
Ensure that all fieldef keys are string to produce a valid error message

### PR DESCRIPTION
**Pull request by benjello**

### CODE REVIEW

File: liam2/entities.py [hunk 1]
```diff
@@ -107,9 +107,12 @@ def from_def(cls, name, fielddef):
                 output = fielddef.pop('output', True)
                 default_value = fielddef.pop('default', default_value_by_strtype[strtype])
                 if fielddef:
-                    invalid_keywords = sorted(fielddef.keys())
-                    raise SyntaxError('invalid keyword(s) found in field definition: %s'
-                                      % ', '.join(invalid_keywords))
+                    invalid_keywords = [str(definition) for definition in sorted(fielddef.keys())]
```
Comment (gdementen) [inline liam2/entities.py]: AFAICT, this line change is not needed, given the switch to .format below which will not break on non-string arguments.

**Review by gdementen — COMMENTED** [review]
Nice catch, thanks! Please also add a changelog entry.

File: liam2/entities.py (hunk 1 above)
Comment (benjello) [reply liam2/entities.py]: Ok I will test that.

Comment (benjello) [reply liam2/entities.py]: Actually it is the join that breaks if provided with a boolean. So the stringification is needed.

Comment (gdementen) [reply liam2/entities.py]: Oh, sorry, I missed the join.

**Comment by benjello** [comment]
@gdementen: should I merge ?

**Comment by gdementen** [comment]
@benjello I would have like a small changelog entry.

**Comment by benjello** [comment]
Ok I will add it.

**Comment by gdementen** [comment]
LGTM, Thanks !

**Pull request merged**